### ISSUE TITLE
Environmental issues

### ISSUE THREAD

**Issue by miraitowa9** [xnyuwg/procnet#2]
Is it possible to publicize the version number of the environment? I'm experiencing a lot of version incompatibility issues.

**Comment by xnyuwg** [comment]
Thank you for your interest.

It took me some time to find my old environment as follows:
torch: 1.11.0
torch-geometric: 2.0.4
transformers: 4.28.1
numpy: 1.20.3
tqdm: 4.62.3

I think using the latest version may generate better results. One potential reason for the problem is that the PyTorch version is lower than the requirement of the latest PyG or Transformers. I think the easiest way is to install the latest version of all of them.
(You may also want to check:
https://pytorch-geometric.readthedocs.io/en/latest/install/installation.html
https://pytorch.org/get-started/locally/)

Hope it works for you.

**Comment by miraitowa9** [comment]
Thank you very much for your reply, the environmental issue I have successfully resolved. Is it convenient to disclose the code of the data processing part of DuEE-fin dataset from your side? Or is it possible to provide a link to process the DuEE-fin dataset? @xnyuwg

**Comment by miraitowa9** [comment]
Hello!&nbsp; Sorry to bother you, I am very interested in your paper "Document-Level Multi-Event Extraction with Event Proxy Nodes and Hausdorff Distance Minimization". Could you please provide the code for the data processing part of the procnet model for the DuEE-fin dataset?







张甜
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "xnyuwg/procnet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年10月1日(星期天) 晚上9:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [xnyuwg/procnet] Environmental issues (Issue #2)





 
Thank you for your interest.
 
It took me some time to find my old environment as follows:
 torch: 1.11.0
 torch-geometric: 2.0.4
 transformers: 4.28.1
 numpy: 1.20.3
 tqdm: 4.62.3
 
I think using the latest version may generate better results. One potential reason for the problem is that the PyTorch version is lower than the requirement of the latest PyG or Transformers. I think the easiest way is to install the latest version of all of them.
 (You may also want to check:
 https://pytorch-geometric.readthedocs.io/en/latest/install/installation.html
 https://pytorch.org/get-started/locally/)
 
Hope it works for you.

**Comment by xnyuwg** [comment]
Glad to hear you solve it.

The process for the DuEE-fin dataset is straightforward. The DuEE-fin dataset is transformed into the ChFinANN format. Subsequently, the data file will be replaced with the DuEE-fin data. Also, modify the schema in "procnet/procnet/data_example/DocEEexample.py" to match the DuEE-fin configuration. Then, the other parts should remain the same. Allow me some time to try to figure out the best way to share this code. I will update you as soon as possible.

**Comment by miraitowa9** [comment]
Thank you very much for your reply, I have another question for you, did you use trigger words when working with the Duefin dataset?

**Comment by xnyuwg** [comment]
I just updated a preliminary version of the code. I just try to upload it as soon as possible to provide you with an overview first.

If you're interested in running with the DuEE_fin, ensure that you've placed the DuEE_fin data in the "/Data/DuEE_fin_Data/" directory (you may need to create it). Unfortunately, seems I cannot directly redistribute this dataset, so you might have to source it independently. Once that's sorted, execute 'python generate_pseudo_Doc2EDAG_dataset.py' to convert the DuEE_fin data to the Doc2EDAG format. Post that, the regular 'bash run.sh' command will suffice, but remember to include '--read_pseudo=True' so that the program will read the pseudo data.

Our method is specifically for non-trigger event extraction and regards DuEE_fin as a non-trigger dataset.

Hope this works for you.

**Comment by miraitowa9** [comment]
Thank you very much for sharing the code! I have another question for you, is the result of the argument extraction the result of the prediction given the standard answer for the event type? If not, are there any results given for event type detection?
Good luck with your research!

**Comment by miraitowa9** [comment]
Predicted results for Duee_Fin：Hello, I would like to generate annotated results for DueeFin test set predictions, could you please give some hints on exactly how I should generate predictions for DueeFin test sets?

**Comment by xnyuwg** [comment]
Thanks for your reply. 

The model is required to predict both argument and event types. None of them are given. From my understanding of the metric code from Doc2EDAG, the evaluation focuses on event types first, where incorrect event types lead to arguments being marked incorrect. Then the evaluation of arguments within the event with correct event types will be done. I think there is no event-type detection evaluation for previous work as well as ours, so I could not provide advice on it.

We did not do experiments on DuEE_Fin test. It is not easy for individuals without a +86 phone number to submit test set results. Therefore, we follow previous research using the dev set of DuEE_Fin for evaluation purposes.

If there's anything more I can assist you with, feel free to reach out.

**Comment by miraitowa9** [comment]
Thank you very much for your reply and good luck with your research!

**Comment by miraitowa9** [comment]
Hello, excuse me, I would like to ask if your code does not have a fixed random seed set? The code I reproduced doesn't give me consistent results every time I run it, and after I set the random seed, it still gives me some differences in the results every time, is there any solution for this?

**Comment by liqiaoshuo** [comment]
> 您好，请问您的代码没有固定的随机种子集吗？我复制的代码每次运行时都不会给出一致的结果，并且在设置随机种子后，每次的结果仍然会出现一些差异，有什么解决方案吗？

您好，我也遇到了同样的问题，请问您是如何解决的呢？

**Comment by xnyuwg** [comment]
Hi,

The randomness was also observed during my experiment. I guess it is CUDA that makes it not easy to completely remove the randomness by setting the seed. The training strategy with Hausdorff could be the main problem that caused such randomness. I think the initialization could be one of the most possible parts that can be improved to reduce the randomness in future work.

According to my observation, there are two kinds of randomness: (1) the convergence epoch varies during training. Sometimes, it converges quickly, while sometimes the score remains at 0.0% for several epochs. I suggest finishing at least 50 epochs for each training and selecting scores in an early stop manner. Do not simply use the results of the last epoch (sorry for the potential unclearness). Based on the code, the simplest way to do it is to use a script to process all the scores stored within each epoch. The reason for storing all results is for the convenience of experimental analysis. Usually, regardless of convergence time, the model will finally achieve a similar score in an early stop way. (2) the score almost lands between 0.825-0.835. That is why I need to run the same training setting several times. I guess improving the initialization, especially the proxy node embeddings, may reduce the randomness, which may be considered future work.

Sorry for the late reply. Thank you for your interest. Hope this could help you. Wishing you smooth sailing in your future research.
